### PR TITLE
feat(google-calendar): show correct sick/vacation message

### PR DESCRIPTION
**Pull request by PatrickDohn**
#### Summary

_Enumerate and describe your changes_

- add logic for if today was marked as sick and a user tries to take a vacation day hrbot will alert user that the day was already marked as sick and vice versa

#### Checklist

_Double check list and complete these actions before merge_

- [x] PR title is in format `TYPE(scope): description`
- [x] Link PR to Issue (if any issues are related)

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const { App, ExpressReceiver } = require('@slack/bolt');
 const _ = require('lodash');
 const stringSimilarity = require('string-similarity');
 const express = require('express');
-
+const moment = require('moment');
```
Comment (davidholyko) [inline src/app.js:5]: add a new line after this

File: src/app.js [hunk 2]
```diff
@@ -279,9 +273,15 @@ app.action(SUBMIT_DATES, async (response) => {
     const startDate = response.body.state.values[key][SELECT_START_DATE]['selected_date'];
     const endDate = response.body.state.values[key][SELECT_END_DATE]['selected_date'];
     const slackName = await WebClientUtil.getUserName(user.id);
-    await googleCalendarUtil.createFutureVacationEvent(slackName, startDate, endDate);
-    const blocks = SlackMessageUtil.buildGenericTextBlock(['Vacation time submitted']);
-    chat.postMessage({ channel, blocks });
+    const today = moment().format('YYYY-MM-DD');
+    if (startDate < today || endDate < startDate) {
```
Comment (davidholyko) [inline src/app.js:277]: this might be better as a util function to validate dates

File: src/app.js [hunk 3]
```diff
@@ -279,9 +273,15 @@ app.action(SUBMIT_DATES, async (response) => {
...
+      const blocks = SlackMessageUtil.buildGenericTextBlock(['Invalid date']);
```
Comment (davidholyko) [inline src/app.js:278]: i think this function can accept a single string as an argument and still work as expecteed

File: src/utils/googleCalendarUtil.js [hunk 4]
```diff
@@ -32,15 +32,25 @@ class GoogleCalendarUtil {
     return response.data.items;
   }
 
-  hasUserCreatedEvent(slackName, events) {
+  // TODO: use this function inside create an event
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:35]: make issue for this

File: src/utils/googleCalendarUtil.js [hunk 5]
```diff
@@ -32,15 +32,25 @@ class GoogleCalendarUtil {
...
+  validateUserEvent(slackName, events) {
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:36]: (optional) add jsDoc for this util function

File: src/app.js (hunk 2 above)
Comment (PatrickDohn) [reply src/app.js]: Updated! Is this what you had in mind?

File: src/app.js (hunk 1 above)
Comment (PatrickDohn) [reply src/app.js:5]: Fixed

File: src/utils/googleCalendarUtil.js (hunk 4 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js:44]: Issue submitted.

**Review by davidholyko — CHANGES_REQUESTED** [review]
remove package lock diff. this PR doesnt add new packages

File: src/utils/googleCalendarUtil.js [hunk 6]
```diff
@@ -32,15 +32,34 @@ class GoogleCalendarUtil {
     return response.data.items;
   }
 
-  hasUserCreatedEvent(slackName, events) {
+  validateDate(startDate, endDate) {
+    const today = moment().format('YYYY-MM-DD');
+    let validatedDate = false;
```
Comment (davidholyko) [inline src/utils/googleCalendarUtil.js:37]: refactor:

```
return startDate < today || endDate < startDate
```

File: src/utils/googleCalendarUtil.js (hunk 5 above)
Comment (PatrickDohn) [reply src/utils/googleCalendarUtil.js:41]: Ill add this when I fix the todo

File: src/app.js [hunk 7]
```diff
@@ -279,9 +275,15 @@ app.action(SUBMIT_DATES, async (response) => {
     const startDate = response.body.state.values[key][SELECT_START_DATE]['selected_date'];
     const endDate = response.body.state.values[key][SELECT_END_DATE]['selected_date'];
     const slackName = await WebClientUtil.getUserName(user.id);
-    await googleCalendarUtil.createFutureVacationEvent(slackName, startDate, endDate);
-    const blocks = SlackMessageUtil.buildGenericTextBlock(['Vacation time submitted']);
-    chat.postMessage({ channel, blocks });
+    const validatedDate = await googleCalendarUtil.validateDate(startDate, endDate);
```
Comment (davidholyko) [inline src/app.js:278]: this function is not async

File: src/utils/slackMessageUtil.js [hunk 8]
```diff
@@ -34,29 +34,17 @@ class SlackMessageUtil {
...
-      blocks.push({
+  buildGenericTextBlock(textInput) {
+    const texts = _.isArray(textInput) ? textInput : [textInput];
+    const blocks = texts.map((text) => {
+      return {
         type: 'section',
         text: {
           type: 'mrkdwn',
-          text,
+          text: text,
```
Comment (davidholyko) [inline src/utils/slackMessageUtil.js:44]: we can do `text,`

File: src/app.js (hunk 7 above)
Comment (PatrickDohn) [reply src/app.js]: Fixed

File: src/utils/slackMessageUtil.js (hunk 8 above)
Comment (PatrickDohn) [reply src/utils/slackMessageUtil.js]: Fixed

**Review by davidholyko — APPROVED** [review]
LGTM

**Pull request merged**